### PR TITLE
feat: add inline type annotation on codegen

### PR DESCRIPTION
**Pull request by joseemds**

### CODE REVIEW

File: packages/ppx/test/css-support/test.expected.re [hunk 1]
```diff
@@ -696,111 +716,125 @@ CssJs.unsafe({js|textCombineUpright|js}, {js| all|js});
...
-CssJs.borderColor(`hsl((`deg(0.), `percent(0.), `percent(0.))));
-CssJs.borderColor(`rgba((0, 51, 178, `percent(0.5))));
-CssJs.borderColor(`rgba((0, 51, 178, `num(0.5))));
-CssJs.borderColor(`rgba((0, 64, 185, `percent(0.5))));
-CssJs.borderColor(`rgba((0, 64, 185, `num(0.5))));
+(CssJs.borderColor(`rgba((0, 51, 178, `percent(0.5)))): CssJs.rule);
+(CssJs.borderColor(`rgba((0, 51, 178, `num(0.5)))): CssJs.rule);
+(CssJs.borderColor(`rgba((0, 64, 185, `percent(0.5)))): CssJs.rule);
+(CssJs.borderColor(`rgba((0, 64, 185, `num(0.5)))): CssJs.rule);
 CssJs.unsafe({js|borderColor|js}, {js| hsla(0 0% 0% /.5)|js});
```
Comment (davesnx) [inline packages/ppx/test/css-support/test.expected.re:803]: unsafe don't have type annotation?

File: packages/ppx/test/css-support/test.expected.re [hunk 2]
```diff
@@ -1776,23 +1818,23 @@ CssJs.unsafe({js|WebkitTextFillColor|js}, {js| $(Color.Text.primary)|js});
 CssJs.unsafe({js|animation|js}, {js| none|js});
 CssJs.unsafe({js|appearance|js}, {js| none|js});
 CssJs.unsafe({js|aspectRatio|js}, {js| 21 / 8|js});
-CssJs.backgroundColor(c);
-CssJs.unsafe({js|bottom|js}, "unset");
-CssJs.boxShadows(`none);
+(CssJs.backgroundColor(c): CssJs.rule);
+(CssJs.unsafe({js|bottom|js}, "unset"): CssJs.rule);
```
Comment (davesnx) [inline packages/ppx/test/css-support/test.expected.re:1822]: but this one have 🤔

File: packages/ppx/src/declarations_to_emotion.re [hunk 3]
```diff
@@ -65,7 +70,7 @@ let render_css_global_values = (name, value) => {
     };
 
   /* bs-css doesn't have those */
-  Ok([[%expr CssJs.unsafe([%e render_string(name)], [%e value])]]);
+  Ok([[%expr (CssJs.unsafe([%e render_string(name)], [%e value]) : CssJs.rule)]]);
```
Comment (davesnx) [inline packages/ppx/src/declarations_to_emotion.re:73]: I see, the ones that bs-css aren't binded. We add the rule.

**Review by davesnx — COMMENTED** [review]
Hey @joseemds. I find it very noisy, to be honest. Since inference should work it out in almost all cases (the only ones that might be beneficial are when there's interpolation).

While adding CSS.Rule and string make %cx extensions extremely noisy and silly.

WDYT?

**Comment by joseemds** [comment]
closing due to #285

**Pull request closed without merging**